rspace/src/main/scala/coop/rchain/rspace/state/RSpaceImporter.scala

```diff
@@ -57,31 +58,17 @@ object RSpaceImporter {
       ).whenA(!sizeIsValid)
     }
 
-    // Validate data items size
-    def validateDataSize[T]: F[Unit] = {
-      val sizeIsValid = receivedDataSize <= chunkSize | isEnd
-      raiseError[T](
-        s"Input size of data items $receivedDataSize is greater then expected chunk size $chunkSize."
-      ).whenA(!sizeIsValid)
-    }
-
-    // Tries decoded from received items
-    // - validate trie hash to match decoded trie hash
-    def tries: F[List[(Blake2b256Hash, Trie)]] = historyItems.toList traverse {
-      case (hash, trieBytes) =>
-        val trie = decodeTrie(trieBytes)
-        // TODO: this is strange that we need to encode Trie again to get the hash.
-        //  - the reason is that hash is not calculated on result of `Trie.codecTrie.encode`
-        //    but on each case of Trie instance (see `Trie.hash`)
-        //  - this adds substantial time for validation e.g. 20k records 450ms (with encoding 2.4sec)
-        // https://github.com/rchain/rchain/blob/4dd216a7/rspace/src/main/scala/coop/rchain/rspace/history/HistoryStore.scala#L25-L26
-        val trieHash = Trie.hash(trie)
-        if (hash == trieHash) (hash, trie).pure[F]
-        else
-          raiseError(
-            s"Trie hash does not match decoded trie, key: ${hash.bytes.toHex}, decoded: ${trieHash.bytes.toHex}."
-          )
-    }
+    // Validate history hashes
+    def validateHistoryItemsHashes: F[List[(ByteVector, ByteVector)]] =
```

Comment:
Better to call this function `getAndValidateHistoryItems` because it's not just validation.